packages/flutter_tools/lib/src/compile.dart

```diff
@@ -302,15 +351,22 @@ class ResidentCompiler {
 
     // First time recompile is called we actually have to compile the app from
     // scratch ignoring list of invalidated files.
+    _PackageUriMapper packageUriMapper;
+    if (request.packagesFilePath != null) {
+      packageUriMapper = _PackageUriMapper(request.mainPath, request.packagesFilePath);
+    }
+
     if (_server == null) {
-      return _compile(_mapFilename(request.mainPath),
-          request.outputPath, _mapFilename(request.packagesFilePath));
+      return _compile(
+          _mapFilename(packageUriMapper, request.mainPath), request.outputPath,
+          _mapFilename(null, request.packagesFilePath)
```

Comment:
can we decorate null with parameter name? `/* packageUriMapper= */ null`, for example?

Reply:
done